src/main/java/carpet/script/language/ControlFlow.java

```diff
@@ -145,5 +151,23 @@ public static void apply(Expression expression) // public just to get the javado
                 return (c_, t_) -> retval;
             }
         });
+
+        expression.addUnaryFunction("encode_b64", v -> StringValue.of(Base64.getEncoder().encodeToString(v.getString().getBytes())));
+        expression.addUnaryFunction("decode_b64", v -> {
+            try {
+                return StringValue.of(new String(Base64.getDecoder().decode(v.getString()), StandardCharsets.ISO_8859_1));//using this charset cos it's the one used in decoding function
```

Comment:
Why not UTF-8?

Reply:
as I mentioned in the comment. that's the charset used in the `Base64.getDecoder.decode(String s)` function, so it makes sense to re-encode with the same charset.

Reply:
well entire scarpet base is in UTF-8 and you are potentially introducing a string with a different base

Reply:
kk, gotcha

Reply:
but you might need to test with some wierd emojis and stuff